package NServiceMonitor.NProto;

message TServiceInstanceInfo {
    optional string ContainerHostname = 1 [json_name = "container_hostname"];
    repeated string Itags = 2 [json_name = "itags"];
}

message TServiceInstanceInfoContainer {
    repeated TServiceInstanceInfo Result = 1 [json_name = "result"];
}

message TServerInfo {
    optional string ControllerStatus = 1 [json_name = "controller_status"];
    optional uint64 ControllerUptime = 2 [json_name = "controller_uptime"];
    optional uint64 CpuCount = 3 [json_name = "cpu_count"];
    optional uint64 CpuLoadSystem = 4 [json_name = "cpu_load_system"];
    optional uint64 CpuLoadUser = 5 [json_name = "cpu_load_user"];
    optional double LoadAverage = 6 [json_name = "load_average"];
    optional string MemSizeReal = 7 [json_name = "mem_size_real"];
    optional string MemSizeVirtual = 8 [json_name = "mem_size_virtual"];
    optional string Product = 9 [json_name = "product"];
    optional uint64 TotalMemSize = 10 [json_name = "total_mem_size"];
    optional string SvnRevision = 11 [json_name = "Svn_revision"];
    optional string SvnRoot = 12 [json_name = "Svn_root"];
    optional string SvnTag = 13 [json_name = "svn_tag"];
}

message TServerInfoContainer {
    optional TServerInfo Result = 1 [json_name = "result"];
}

message TResourceRequests {
    optional uint64 VcpuLimit = 1 [json_name = "vcpu_limit"];
    optional uint64 MemoryLimit = 2 [json_name = "memory_limit"];
}

message TInstanceInfo {
    optional string Name = 1 [json_name = "name"];
    optional string ShortName = 2 [json_name = "short_name"];
    optional TServerInfo ServerInfo = 10 [json_name = "server_info"];
    optional TResourceRequests ResourceRequests = 11 [json_name = "resource_requests"];
};

message TDataCenterFrontendInfo {
    optional string Name = 1 [json_name = "name"];
    repeated TInstanceInfo Instances = 10 [json_name = "instances"];
}

message TServiceFrontendInfo {
    optional string Name = 1 [json_name = "name"];
    repeated TDataCenterFrontendInfo DataCenters = 10 [json_name = "data_centers"];
}

message TServiceFrontendInfoContainer {
    repeated TServiceFrontendInfo Services = 1 [json_name = "services"];
}

message THeader {
    optional string Name = 1;
    optional string Value = 2;
}

message TReplyInfo {
    optional bool IsConnected = 1;
    optional bool IsCorrectReply = 2;
    optional uint32 Code = 3;
    optional string ErrorMessage = 4;
    optional string Content = 5;
    repeated string Flags = 6;
    repeated THeader Headers = 7;
}
